src/Nethermind/Nethermind.Consensus/Processing/BlockchainProcessor.cs

```diff
@@ -480,6 +480,11 @@ private void FireProcessingQueueEmpty()
             Metrics.RecoveryQueueSize = Math.Max(_queueCount - blockQueueCount - (IsProcessingBlock ? 1 : 0), 0);
             Metrics.ProcessingQueueSize = blockQueueCount;
             _stats.UpdateStats(lastProcessed, processingBranch.Root, blockProcessingTimeInMicrosecs);
+
+            foreach (Block block in processedBlocks)
+            {
+                _blockTree.CacheBlockNumber(block.Hash, block.Header.Number);
```

Comment:
Do we need explicit call? Most of the processed blocks should be recently inserted, and for restarts we can think of something better.

Reply:
Oh, get what you are saying. Let me check

Reply:
Yes for some reason (looking into this too) 

<img width="655" alt="image" src="https://github.com/user-attachments/assets/e917bf5c-8ddd-417a-b881-fca98035b3fb" />

Reply:
GetTotalDifficulty need to die.

Reply:
If you want the full path 
<img width="581" alt="image" src="https://github.com/user-attachments/assets/c8ecd7f9-3a27-4296-ad40-375c67aef2d2" />